operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentConfig.java

```diff
@@ -0,0 +1,266 @@
+                            .withValue(Constants.CERTIFICATES_FOLDER + "/tls.crt")
+                            .build(),
+                    new EnvVarBuilder()
+                            .withName("KC_HTTPS_CERTIFICATE_KEY_FILE")
+                            .withValue(Constants.CERTIFICATES_FOLDER + "/tls.key")
+                            .build(),
+                    new EnvVarBuilder()
+                            .withName("KC_PROXY")
+                            .withValue("passthrough")
+                            .build());
+
+            envVars.addAll(enabledTls);
+
+            var volume = new VolumeBuilder()
+                    .withName("keycloak-tls-certificates")
+                    .withNewSecret()
+                    .withSecretName(tlsSecret)
+                    .withOptional(false)
+                    .endSecret()
+                    .build();
+
+            var volumeMount = new VolumeMountBuilder()
+                    .withName(volume.getName())
+                    .withMountPath(Constants.CERTIFICATES_FOLDER)
+                    .build();
+
+            deployment.getSpec().getTemplate().getSpec().getVolumes().add(volume);
+            kcContainer.getVolumeMounts().add(volumeMount);
+        }
+
+        var userRelativePath = readConfigurationValue(Constants.KEYCLOAK_HTTP_RELATIVE_PATH_KEY);
+        var kcRelativePath = (userRelativePath == null) ? "" : userRelativePath;
+        var protocol = (keycloakCR.getSpec().isHttp()) ? "http" : "https";
+        var kcPort = (keycloakCR.getSpec().isHttp()) ? Constants.KEYCLOAK_HTTP_PORT : Constants.KEYCLOAK_HTTPS_PORT;
+
+        var baseProbe = new ArrayList<>(List.of("curl", "--head", "--fail", "--silent"));
+
+        if (!keycloakCR.getSpec().isHttp()) {
+            baseProbe.add("--insecure");
+        }
+
+        var readyProbe = new ArrayList<>(baseProbe);
+        readyProbe.add(protocol + "://127.0.0.1:" + kcPort + kcRelativePath + "/health/ready");
+        var liveProbe = new ArrayList<>(baseProbe);
+        liveProbe.add(protocol + "://127.0.0.1:" + kcPort + kcRelativePath + "/health/live");
+
+        kcContainer
+                .getReadinessProbe()
+                .setExec(new ExecActionBuilder().withCommand(readyProbe).build());
+        kcContainer
+                .getLivenessProbe()
+                .setExec(new ExecActionBuilder().withCommand(liveProbe).build());
+    }
+
+    public void configureFeatures() {
+        var featureSpec = keycloakCR.getSpec().getFeatureSpec();
+        if (featureSpec == null) return;
+
+        var kcContainer = deployment.getSpec().getTemplate().getSpec().getContainers().get(0);
+        var args = kcContainer.getArgs();
+        var enabledFeatures = featureSpec.getEnabledFeatures();
+        var disabledFeatures = featureSpec.getDisabledFeatures();
+
+        if (CollectionUtil.isNotEmpty(enabledFeatures)) {
+            args.add("--features=" + CollectionUtil.join(enabledFeatures, ","));
```

Comment:
@vmuzikar I think it's not possible to specify features by env var, right?

Reply:
Of course it is. 🙂 We should not be setting arguments but rather env vars. See https://www.keycloak.org/server/all-config#_feature

Reply:
Oh, I see. I thought it was stored in some separate config afterwards. Thanks!